### PR TITLE
Fix: prestashop event functions are overwritten in the "classic" theme

### LINKED ISSUE #27473: Bug of the events functions on the prestashop variable (in core.js)
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

If we want to access `prestashop.selectors` , the logic would be to wait for the `'selectorsInit'` event. So I naturally created a listener:

```jsx
prestashop.on('selectorsInit', () => {
...
});
```
Except that the event never triggers.

### Expected behavior

The event should fire on page load

### Steps to reproduce

Create an event JavaScript in a module or in the theme:
```jsx
prestashop.on('selectorsInit', () => {
...
});
```

### PrestaShop version(s) where the bug happened

1.7.8.2

### PHP version(s) where the bug happened

ALL

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by tups**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fix event handling on `prestashop` variable
| Type?             | bug fix
| Category?         | FO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27473
| How to test?      | Create an event JavaScript in a module or in the theme
| Possible impacts? | Prestashop theme events

Create an event JavaScript in a module or in the theme:
```jsx
prestashop.on('selectorsInit', () => {
	...
});
```

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27475)

### CODE REVIEW

File: themes/_core/js/product.js [hunk 1]
```diff
@@ -54,10 +54,10 @@ function getProductUpdateUrl() {
...
   if (
     prestashop !== null
-    && prestashop.urls !== null
-    && prestashop.urls.pages !== null
-    && prestashop.urls.pages.product !== ''
-    && prestashop.urls.pages.product !== null
+        && prestashop.urls !== null
+        && prestashop.urls.pages !== null
+        && prestashop.urls.pages.product !== ''
+        && prestashop.urls.pages.product !== null
```
Comment (NeOMakinG) [inline themes/_core/js/product.js]: Hello,

These tabs are really unnecessary, right? Same for every other next

Comment (tups) [reply themes/_core/js/product.js]: Hello,
I just checked, indeed it's a strange, automatic formatting that has trouble happening. I just corrected it.

**Comment by tups** [comment]
I just made the changes seen with @NeOMakinG
This change fixes overwriting of `emit` and `on` functions of `prestashop` object from theme

File: themes/_core/js/theme.js [hunk 2]
```diff
@@ -1,8 +1,5 @@
 import $ from 'jquery';
 
-import prestashop from 'prestashop';
-// eslint-disable-next-line
-import EventEmitter from "events";
```
Comment (NeOMakinG) [inline themes/_core/js/theme.js]:
```suggestion
```

**Review by NeOMakinG — APPROVED** [review]

**Review by matks — COMMENTED** [review]
@NeOMakinG @tups I'm a bit puzzled by this code 🤔 .

I see the issue, I understand the bugfix. But what I dont understand is: why was the purpose of the code before? Do you know?

I just like to know what we remove to be fully aware of the consequences of this removal

**Comment by NeOMakinG** [comment]
> @NeOMakinG @tups I'm a bit puzzled by this code 🤔 .
> 
> I see the issue, I understand the bugfix. But what I dont understand is: why was the purpose of the code before? Do you know?
> 
> I just like to know what we remove to be fully aware of the consequences of this removal

Sincerely I don't know, I think it's just a mistake

**Comment by matks** [comment]
@NeOMakinG so we going to do a gamble ? 😄 

🎲

**Review by matks — APPROVED** [review]
Well, if we break something important we should observe that when we build the beta. So let's follow this :)

**Comment by kpodemski** [comment]
This was introduced years ago by @djfm 
https://github.com/PrestaShop/PrestaShop/commit/196b35d451d8a4b011b9740d4fb0d2b01c35c14a

Unfortunately, I'm not sure what was the idea behind this code. @Oksydan, any clues?

**Comment by djfm** [comment]
Wow long time I had not been involved in a PrestaShop issue :)

I believe your fix is correct.

At the time of the commit quoted by @kpodemski the global `prestashop` object did not extend EventEmitter and if we had to monkey patch it like that it was probably because there was no easy way to change how it was instantiated, I don't remember the details. But since we were sure that the monkey patch file was included before any theme related JS ran, it was, at the time, safe.

I don't have a computer with me and searching on GitHub on a phone is not very practical: where is that global `prestashop` object instantiated now?

I'm guessing it is now a proper event emitter and the monkey patch happens after some handlers have already been defined, effectively removing them.

So yes I'm pretty sure removing that "inherit" is the right move.

If I'm wrong feel free not to pay me this month :)

**Comment by NeOMakinG** [comment]
> Wow long time I had not been involved in a PrestaShop issue :)
> 
> I believe your fix is correct.
> 
> At the time of the commit quoted by @kpodemski the global `prestashop` object did not extend EventEmitter and if we had to monkey patch it like that it was probably because there was no easy way to change how it was instantiated, I don't remember the details. But since we were sure that the monkey patch file was included before any theme related JS ran, it was, at the time, safe.
> 
> I don't have a computer with me and searching on GitHub on a phone is not very practical: where is that global `prestashop` object instantiated now?
> 
> I'm guessing it is now a proper event emitter and the monkey patch happens after some handlers have already been defined, effectively removing them.
> 
> So yes I'm pretty sure removing that "inherit" is the right move.
> 
> If I'm wrong feel free not to pay me this month :)

It's instantiated in some JS from the core, which is loaded first, except if as a theme developer you decided to change the order

Implementing a new event emitter in the theme override the core event emitter, so you are not able to use some core's events

I did the QA on the PR and helped @tups finding this, should be fine honestly :)

**Review by kpodemski — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @tups,

I tried to install your theme with PS1783
[classic-child.zip](https://github.com/PrestaShop/PrestaShop/files/8117813/classic-child.zip)
But I don't have any errors in the console
I attached a screen record

https://user-images.githubusercontent.com/16067358/155162614-f21861ad-4a74-404f-87fd-f0ad0e844eaa.mp4

Could you please provide us with a demo module or a demo theme to reproduce the issue?

Thanks!

**Comment by kpodemski** [comment]
@PrestaShop/prestashop-maintainers 

I think it should be validated by dev?

**Comment by NeOMakinG** [comment]
I just tested, it, works fine

- Before the PR => no `selectorsInit` because the `EventEmitter` is erased by the theme, and you can't even wait for page load because your asset is after the `core.js` file
- After the PR => you can just use `prestashop.addEventListener('selectorsInit', ...)` without document.ready, and you will be able to catch events thrown by the core on page ready

**Comment by NeOMakinG** [comment]
Thanks @tups

**Comment by PrestaEdit** [comment]
Hi everyone,

Could you confirm me that it's working? 

My `on()`/`once()` event seems not handled.
Also, the `addEventListener` is not a available function of `prestashop` object.

**Pull request merged**